packages/@aws-cdk/aws-apigateway/lib/authorizers/cognito.ts

```diff
@@ -96,6 +103,14 @@ export class CognitoUserPoolsAuthorizer extends Authorizer implements IAuthorize
     }
 
     this.restApiId = restApi.restApiId;
+
+    const deployment = restApi.latestDeployment;
+    if (deployment) {
+      deployment.node.addDependency(this);
+      deployment.addToLogicalId({
+        authorizer: this.authorizerProps,
+      });
```

Comment:
As I understand it, the dependency just ensures resource deployment order, and it won't force an update to another resource. We really need a test to ensure that whenever the lambda function changes, the construct that needs to be updated is actually updated.

Reply:
I've made a change and added a new test which checks that the deployment changes when the lambda function changes name